internal/engine/build_and_deployer.go

```diff
@@ -66,15 +67,28 @@ func (composite *CompositeBuildAndDeployer) BuildAndDeploy(ctx context.Context,
 	return BuildResult{}, lastErr
 }
 
+func isPermanentError(err error) bool {
+	if _, ok := err.(*model.ValidateErr); ok {
+		return true
+	}
+
+	cause := errors.Cause(err)
```

Comment:
is this robust against fmt.errorf-ing on top of a context.Canceled error?

Reply:
no, that's why we use errors.Wrap! :raised_hands: